### DISCUSSION TITLE
Vercel automatically putting node_env = production on deployment?

### DISCUSSION THREAD

**Discussion by mercpls** [vercel/community#843] [Help] [answered]
Does vercel automatically place the node_env to production? I'm having difficulty getting my environment to set to dev when deploying. 

Do I have to setup a separate domain and using a custom env key that isn't NODE_ENV?

**Comment by amyegan** [comment, marked as answer]
Hi @mercpls! If you're trying to get the current build environment, you can use the built in [System Environment Variables](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables) provided by Vercel. `VERCEL_ENV` is provided automatically and will give you the current environment. And there are many framework prefixes to help you expose Environment Variables to the browser, such as `NEXT_PUBLIC_VERCEL_ENV`.
